tests/test_leaderboard_line_writer.py

```diff
@@ -0,0 +1,75 @@
+        self.assertEqual(self.leaderboard_line_writer.background_rect_height, self.background_rect_height)
+        self.assertEqual(self.leaderboard_line_writer.leaderboard_title_rect_height, self.leaderboard_title_rect_height)
+
+    def test_construtor_empty_list(self):
+        empty_leaderboard_line_writer = LeaderboardLineWriter("", self.background_rect_width,
+                                                              self.background_rect_height,
+                                                              self.leaderboard_title_rect_height)
+        self.assertEqual(empty_leaderboard_line_writer.leaderboard_score_lines, [])
+        self.assertEqual(empty_leaderboard_line_writer.line, 0)
+        # Because default width is 500
+        self.assertEqual(empty_leaderboard_line_writer.width, 500 + 100)
+        self.assertEqual(empty_leaderboard_line_writer.line_width, 500)
+        # 30 because default font size is 24, and line height is font size + 6
+        self.assertEqual(empty_leaderboard_line_writer.line_height, 30)
+        self.assertEqual(empty_leaderboard_line_writer.size, 24)
+        self.assertEqual(empty_leaderboard_line_writer.leaderboard_scores_text, "")
+        self.assertEqual(empty_leaderboard_line_writer.color, 'white')
+        self.assertEqual(type(empty_leaderboard_line_writer.font), type(pygame.font.SysFont('ariel', 24)))
+        self.assertEqual(empty_leaderboard_line_writer.background_rect_width, self.background_rect_width)
+        self.assertEqual(empty_leaderboard_line_writer.background_rect_height, self.background_rect_height)
+        self.assertEqual(empty_leaderboard_line_writer.leaderboard_title_rect_height,
+                         self.leaderboard_title_rect_height)
+
+    def test_update(self):
+        #Make sure render_all_lines is called
+        self.leaderboard_line_writer.render_all_lines = MagicMock()
+        self.leaderboard_line_writer.update()
+        self.leaderboard_line_writer.render_all_lines.assert_called_once()
+
+    def test_render_all_lines(self):
+        self.leaderboard_line_writer.render_score_line = MagicMock()
+        self.leaderboard_line_writer.update()
+        calls = [call(line) for line in self.leaderboard_line_writer.leaderboard_scores_text.splitlines()]
+        self.leaderboard_line_writer.render_score_line.assert_has_calls(calls, any_order=False)
+
+    def test_render_score_line(self):
+        lines = self.leaderboard_line_writer.leaderboard_scores_text.splitlines()
+        old_line_num = self.leaderboard_line_writer.line
+        old_leaderboard_score_lines_len = len(self.leaderboard_line_writer.leaderboard_score_lines)
+        self.leaderboard_line_writer.render_score_line(lines[0])
+        self.assertEqual(self.leaderboard_line_writer.line,old_line_num+1)
+        self.assertEqual(len(self.leaderboard_line_writer.leaderboard_score_lines), old_leaderboard_score_lines_len+1)
+
```

Comment:
Wow, great work thinking up so many test cases!

Reply:
Thanks! I tried to do as much as I could but it was hard to test!